app/controllers/deliveries/new/select-project.js

```diff
@@ -0,0 +1,62 @@
+import Ember from 'ember';
+
+const PROJECT_ADMIN_AUTH_ROLE = 'project_admin';
+
+export default Ember.Controller.extend({
+  project: null,
+  application: Ember.inject.controller(),
+  currentDukeDsUser: Ember.computed.alias('application.currentDukeDsUser'),
+  currentUserProjectPermissions: Ember.computed('project', 'currentDukeDsUser.id', function () {
```

Comment:
I see several instances where computed properties are returning promises (e.g. the result of a `store.query()`), which [can make you want to tear your hair out](https://emberigniter.com/guide-promises-computed-properties/). I know it's given me a lot of grief during development and testing, and I'm guessing it's why these functions have so many early bail outs.

I'd encourage you to have a look at that article to see if any of those patterns apply, and I'll mention that I've had good results with setting the property after the promise resolves:

1. For controllers, you can add an observer on init (or some other property). This is how the application controller sets the current user:

https://github.com/Duke-GCB/datadelivery-ui/blob/86edec2b4768336c28dbf6c8c606ece176cf4c7f/app/controllers/application.js#L7-L16

2. For components, kick off the promises in `didInsertElement()`, since they don't have the same object lifecycle as other objects (via https://emberigniter.com/render-promise-before-it-resolves/):

https://github.com/Duke-GCB/bespin-ui/blob/9d25758fb6d82c7e46f309bee248877b239685c8/app/components/questionnaire/dds-project-field.js#L12-L21

Reply:
That guide says that there is a danger with this approach (`#1` in the guide):
> waiting for promises and this.setting – on a potentially destroyed component

Reply:
I'm going to use the `#3 Special promise-backed objects` option.